components/properties/src/bidi_data.rs

```diff
@@ -0,0 +1,204 @@
+/// struct is `None`.
+#[derive(Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub struct BidiMirroringProperties {
+    /// Represents the Bidi_Mirroring_Glyph property value
+    pub mirroring_glyph: Option<char>,
+    /// Represents the Bidi_Mirrored property value
+    pub mirrored: bool,
+}
+
+/// The enum represents Bidi_Paired_Bracket_Type, the char represents Bidi_Paired_Bracket.
+/// Bidi_Paired_Bracket has a value of `None` when Bidi_Paired_Bracket_Type is `None`.
+#[derive(Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum BidiPairingProperties {
+    /// Represents Bidi_Paired_Bracket_Type=Open, and the Bidi_Paired_Bracket value for that code point.
+    Open(char),
+    /// Represents Bidi_Paired_Bracket_Type=Close, and the Bidi_Paired_Bracket value for that code point.
+    Close(char),
+    /// Represents Bidi_Paired_Bracket_Type=None, which cooccurs with Bidi_Paired_Bracket
+    /// being undefined for that code point.
+    None,
+}
+
+/// A borrowed wrapper around Bidi properties data, returned by
+/// [`BidiAuxiliaryProperties::as_borrowed()`]. More efficient to query.
+#[derive(Debug)]
+pub struct BidiAuxiliaryPropertiesBorrowed<'a> {
+    data: &'a BidiAuxiliaryPropertiesV1<'a>,
+}
+
+impl<'a> BidiAuxiliaryPropertiesBorrowed<'a> {
+    fn convert_mirroring_glyph_data(trie_data_char: char) -> Option<char> {
+        if trie_data_char as u32 == 0 {
+            None
+        } else {
+            Some(trie_data_char)
+        }
+    }
+
+    /// Return a struct for the given code point representing Bidi mirroring-related
+    /// property values. See [`BidiMirroringProperties`].
+    ///
+    /// # Examples
+    /// ```
+    /// use icu_properties::{bidi_data, bidi_data::BidiMirroringProperties};
+    ///
+    /// let data =
+    ///     bidi_data::load_bidi_auxiliary_properties_unstable(&icu_testdata::unstable())
+    ///         .expect("The data should be valid");
+    /// let bidi_data = data.as_borrowed();
+    ///
+    /// let open_paren = bidi_data.get32_mirroring_props('(' as u32);
+    /// assert_eq!(open_paren.mirroring_glyph, Some(')'));
+    /// assert_eq!(open_paren.mirrored, true);
+    /// let close_paren = bidi_data.get32_mirroring_props(')' as u32);
+    /// assert_eq!(close_paren.mirroring_glyph, Some('('));
+    /// assert_eq!(close_paren.mirrored, true);
+    /// let open_angle_bracket = bidi_data.get32_mirroring_props('<' as u32);
+    /// assert_eq!(open_angle_bracket.mirroring_glyph, Some('>'));
+    /// assert_eq!(open_angle_bracket.mirrored, true);
+    /// let close_angle_bracket = bidi_data.get32_mirroring_props('>' as u32);
+    /// assert_eq!(close_angle_bracket.mirroring_glyph, Some('<'));
+    /// assert_eq!(close_angle_bracket.mirrored, true);
+    /// let three = bidi_data.get32_mirroring_props('3' as u32);
+    /// assert_eq!(three.mirroring_glyph, None);
+    /// assert_eq!(three.mirrored, false);
+    /// ```
+    pub fn get32_mirroring_props(&self, code_point: u32) -> BidiMirroringProperties {
```

Comment:
suggestion (nb): If we do not plan to have char-accepting methods, do we need to call this `get32`?

(There's not much point to having char-accepting methods since they won't be faster and the caller can `as u32` when needed)

Reply:
I suggested get32 because I think we should also have char methods